app/src/main/java/de/aurora/mggvertretungsplan/MainActivity.java

```diff
@@ -220,6 +221,68 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case R.id.action_feedback:
                 launchCustomTabsIntent(color, getString(R.string.feedback_url));
                 break;
+            case R.id.action_share:
+                // Share
+                String shareText = "Vertretungsplan " + class_name;
+                String input = StorageUtilities.readFile(this);
+                try {
+                    JSONArray json = new JSONArray(input);
+                    // For every Week:
+                    for (int i = 0; i < json.length(); i++) {
```

Comment:
Here you are looping through all of the json elements, which is hell of a work. Since I don't want duplicate code, you can instantiate a TimeTable instance by calling `TimeTable(json);`.